### PR TITLE
[Break XPU] [Inductor] Refine triton_bundler.py to support correctly on Intel GPU and fix CI failures.

### PR DESCRIPTION
**Pull request by etaf**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #139872
* __->__ #139705

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang @aakhundov

### CODE REVIEW

File: torch/_inductor/triton_bundler.py [hunk 1]
```diff
@@ -231,8 +233,9 @@ def read_and_emit(
                             )
                         file.write(payload)
                     counters["inductor"]["triton_bundler_read_and_emit_kernel"] += 1
-                    if artifact.filename.endswith(".cubin"):
-                        # Each kernel has bunch of files like .cubin, .json, .ttir
+                    file_extension = os.path.splitext(artifact.filename)
```
Comment (guangyey) [inline torch/_inductor/triton_bundler.py:236]:
```suggestion
                    _, file_extension = os.path.splitext(artifact.filename)
```

File: torch/_inductor/utils.py [hunk 2]
```diff
@@ -90,6 +90,7 @@ def get_gpu_type():
 VarRanges = Dict[sympy.Expr, sympy.Expr]
 InputType = Optional[Union[torch.Tensor, int, torch.SymInt]]
 
+GPU_KERNEL_BINS = {"cuda": "cubin", "xpu": "spv"}
```
Comment (guangyey) [inline torch/_inductor/utils.py:93]:
```suggestion
GPU_KERNEL_BIN_EXTS = {"cuda": ".cubin", "xpu": ".spv"}
```

File: torch/_inductor/triton_bundler.py (hunk 1 above)
Comment (guangyey) [reply torch/_inductor/triton_bundler.py:236]: `os.path.splitext` will return a tuple including `root` and `ext`, refer to https://docs.python.org/3/library/os.path.html#os.path.splitext

File: torch/_inductor/utils.py (hunk 2 above)
Comment (guangyey) [reply torch/_inductor/utils.py:93]: As doc said, `ext` will contain a period.

Comment (etaf) [reply torch/_inductor/utils.py]: Resolved, thanks for your kindly reminder.

**Review by EikanWang — APPROVED** [review]

File: torch/_inductor/triton_bundler.py [hunk 3]
```diff
@@ -159,8 +160,10 @@ def collect(
                             counters["inductor"]["triton_bundler_save_kernel"] += 1
                         except Exception:
                             log.debug("failed to collect triton kernel", exc_info=True)
-                        if filename.endswith(".cubin"):
-                            # Each kernel has bunch of files like .cubin, .json, .ttir
+                        file_extension = os.path.splitext(filename)[1]
+                        if file_extension in GPU_KERNEL_BIN_EXTS.values():
+                            print(file_extension)
```
Comment (jansel) [inline torch/_inductor/triton_bundler.py:165]: remove

**Review by jansel — CHANGES_REQUESTED** [review]

File: torch/_inductor/utils.py [hunk 4]
```diff
@@ -90,6 +90,7 @@ def get_gpu_type():
 VarRanges = Dict[sympy.Expr, sympy.Expr]
 InputType = Optional[Union[torch.Tensor, int, torch.SymInt]]
 
+GPU_KERNEL_BIN_EXTS = {"cuda": "cubin", "xpu": "spv"}
```
Comment (guangyey) [inline torch/_inductor/utils.py:93]:
```suggestion
GPU_KERNEL_BIN_EXTS = {"cuda": ".cubin", "xpu": ".spv"}
```
You are missing a period

File: torch/_inductor/triton_bundler.py [hunk 5]
```diff
@@ -231,8 +234,9 @@ def read_and_emit(
                             )
                         file.write(payload)
                     counters["inductor"]["triton_bundler_read_and_emit_kernel"] += 1
-                    if artifact.filename.endswith(".cubin"):
-                        # Each kernel has bunch of files like .cubin, .json, .ttir
+                    file_extension = os.path.splitext(artifact.filename)[1]
+                    if file_extension in GPU_KERNEL_BIN_EXTS.values():
```
Comment (guangyey) [inline torch/_inductor/triton_bundler.py:238]: `GPU_KERNEL_BIN_EXTS` misses a period resulting in this branch never matching.

File: torch/_inductor/triton_bundler.py [hunk 6]
```diff
@@ -159,8 +160,10 @@ def collect(
                             counters["inductor"]["triton_bundler_save_kernel"] += 1
                         except Exception:
                             log.debug("failed to collect triton kernel", exc_info=True)
-                        if filename.endswith(".cubin"):
-                            # Each kernel has bunch of files like .cubin, .json, .ttir
+                        file_extension = os.path.splitext(filename)[1]
+                        if file_extension in GPU_KERNEL_BIN_EXTS.values():
```
Comment (guangyey) [inline torch/_inductor/triton_bundler.py:164]: GPU_KERNEL_BIN_EXTS misses a period resulting in this branch never matching.

**Review by guangyey — CHANGES_REQUESTED** [review]
GPU_KERNEL_BIN_EXTS misses a period resulting in this branch never matching.

**Review by EikanWang — CHANGES_REQUESTED** [review]
Pls. address the comments.

File: torch/_inductor/triton_bundler.py (hunk 6 above)
Comment (EikanWang) [reply torch/_inductor/triton_bundler.py:164]: @etaf , why could the logic pass the test cases?

**Review by EikanWang — CHANGES_REQUESTED** [review]

File: torch/_inductor/utils.py (hunk 4 above)
Comment (etaf) [reply torch/_inductor/utils.py]: Refined, thanks.

File: torch/_inductor/triton_bundler.py (hunk 3 above)
Comment (etaf) [reply torch/_inductor/triton_bundler.py]: Removed, thanks.

**Comment by EikanWang** [comment]
> @EikanWang The file_extension part is currently not reachable, so I can't create case to cover them.

Could you help me understand why the code currently is not enabled?

**Review by EikanWang — APPROVED** [review]
LGTM now.

**Review by jansel — APPROVED** [review]

**Review by guangyey — APPROVED** [review]

**Pull request closed without merging**